lib/tenstorrent/bh_chip/bh_chip.c

```diff
@@ -147,7 +147,55 @@ int therm_trip_gpio_setup(struct bh_chip *chip)
 	}
 	gpio_init_callback(&chip->therm_trip_cb, therm_trip_detected,
 			   BIT(chip->config.therm_trip.pin));
-	ret = gpio_add_callback(chip->config.therm_trip.port, &chip->therm_trip_cb);
+	ret = gpio_add_callback_dt(&chip->config.therm_trip, &chip->therm_trip_cb);
+
+	return ret;
+}
+
+void pgood_fault_detected(const struct device *dev, struct gpio_callback *cb, uint32_t pins)
+{
+	static const struct gpio_dt_spec board_fault_led =
+		GPIO_DT_SPEC_GET_OR(DT_PATH(board_fault_led), gpios, {0});
+	struct bh_chip *chip = CONTAINER_OF(cb, struct bh_chip, pgood_cb);
+
+	/* Assert board fault */
+	gpio_pin_set_dt(&board_fault_led, 1);
+	/* Report over SMBus - to add later */
```

Comment:
Do we want to merge this without SMBUS reporting, or is this PR more a WIP right now?